test/test_uri_parser.py

```diff
@@ -521,6 +527,34 @@ def test_special_chars(self):
         self.assertEqual(user, res["username"])
         self.assertEqual(pwd, res["password"])
 
+    def test_detect_external_db(self):
+        hosts = [
+            "normalhost.com",
+            "host.cosmos.AZURE.com",
+            "host.docdb.amazonaws.com",
+            "host.docdb-elastic.amazonaws.com",
+        ]
+        with self.assertLogs("pymongo", level="INFO") as cm:
```

Comment:
How does `self.assertLogs("pymongo", level="INFO") `? Seems super useful!

Reply:
Here's the official docs: https://docs.python.org/3.11/library/unittest.html#unittest.TestCase.assertLogs. It's really helpful for testing our logging behavior, especially now that we have several components that log.